bundler/lib/bundler/plugin/index.rb

```diff
@@ -71,6 +71,15 @@ def register_plugin(name, path, load_paths, commands, sources, hooks)
         raise
       end
 
+      def remove_plugin(name)
+        @commands.delete_if {|_, v| v == name }
```

Comment:
`@commands` looks like the same kind of structure as `@sources`, so it should be removed with the following I guess?
 
```suggestion
        @commands.delete(name)
```

Reply:
I do not remember the exact structure of plugins index file right now, but here we should remove by matching value (as opposed to `delete` method, which deletes matching key).